src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.FrameworkReferenceResolution.targets

```diff
@@ -431,7 +431,7 @@ Copyright (c) .NET Foundation. All rights reserved.
       <_ExistingReferenceAssembliesPackageReference Include="@(PackageReference)" Condition="'%(PackageReference.Identity)' == 'Microsoft.NETFramework.ReferenceAssemblies'"/>
 
       <PackageReference Include="Microsoft.NETFramework.ReferenceAssemblies" Version="$(MicrosoftNETFrameworkReferenceAssembliesLatestPackageVersion)" IsImplicitlyDefined="true" PrivateAssets="All"
-                        Condition="'$(_FullFrameworkReferenceAssemblyPaths)' == '' and '@(_ExistingReferenceAssembliesPackageReference)' == ''"/>
+                        Condition="('$(_FullFrameworkReferenceAssemblyPaths)' == '' or $(_FullFrameworkReferenceAssemblyPaths.Contains('microsoft.netframework.referenceassemblies'))) and '@(_ExistingReferenceAssembliesPackageReference)' == ''"/>
```

Comment:
I don't like this, but I can't see a better way.

I was hoping that we could add the PackageReference with some special metadata `AddedByIncludeTargetingPackReference`, and then check on that in the filter, but it doesn't work, because the subsequent evaluation (with the package ref included) doesn't actually have the PackageReference item, just its consequences through `project.assets.json`.

Reply:
Instead of doing a string contains check on a full path, would it be possible to check whether there are any `ResolvedCompileFileDefinitions` items (which come from `ResolvePackageAssets`) with `NuGetPackageId` metadata that starts with "microsoft.netframework.referenceassemblies"?

Reply:
@dsplaisted I tried this idea today and couldn't get it to work. I think the reason is that adding `AfterTargets="ResolvePackageAssets"` pushes this target too late to add another package reference successfully, so we're back to having a restore loop.

I cannot see a better approach forward for this PR than the `Contains` check unfortunately. Thankfully the string we're looking for is quite unique.

As an aside, the .NET Project System is supposed to detect such restore loops and that's clearly not working as restores in VS run indefinitely, burning CPU.

Reply:
> As an aside, the .NET Project System is supposed to detect such restore loops and that's clearly not working as restores in VS run indefinitely, burning CPU.

Tracking this in https://github.com/dotnet/project-system/issues/9050. I'm working on a fix now.

Reply:
The VS restore cycle detection has been fixed in https://github.com/dotnet/project-system/pull/9051.